### PR TITLE
Add calc_sat_xx and calc_navigation_measurement errors

### PR DESCRIPTION
**Pull request by rge-exafore**

### CODE REVIEW

**Comment by ljbade** [comment]
Looks good but Travis failed

**Comment by jacobmcnamee** [comment]
Requires https://github.com/swift-nav/libswiftnav/pull/345

File: src/solution.c [hunk 1]
```diff
@@ -445,8 +445,12 @@ static void solution_thread(void *arg)
     }
 
     ephemeris_lock();
-    calc_navigation_measurement(n_ready, p_meas, p_nav_meas,
-                                (double)((u32)nav_tc)/SAMPLE_FREQ, p_e_meas);
+    if (calc_navigation_measurement(n_ready, p_meas, p_nav_meas,
+                                    (double)((u32)nav_tc)/SAMPLE_FREQ, p_e_meas)
+        != 0) {
+      log_error("calc_navigation_measurement() returned an error");
+      continue;
```
Comment (jacobmcnamee) [inline src/solution.c]: This is going to break in a fairly glorious manner when the ephemeris is not unlocked :).

**Comment by jacobmcnamee** [comment]
I think we should also add checking for `calc_sat_state()`, `calc_sat_az_el()`, `calc_sat_doppler()`, and the analogous almanac functions, which all appear to be used in the firmware (but not internally within libswiftnav).

**Comment by rge-exafore** [comment]
All calc_sat_xxx function calls were modified. In simulator I added assert. In other places error handling.
Bug in solution.c is fixed.

**Comment by rge-exafore** [comment]
HITL test: [link](http://sbp-log-analysis.swiftnav.com/#/d/0/q/x/firmware/y/metric/f/firmware/sv/piksi_firmware_v0.21-205-g761aba4.hex/st/1?_k=4omv28)

File: src/simulator.c [hunk 2]
```diff
@@ -278,19 +279,22 @@ void simulation_step_tracking_and_observations(double elapsed)
   /* First we calculate all the current sat positions, velocities */
   for (u8 i=0; i<simulation_num_almanacs; i++) {
     double clock_err, clock_rate_err;
-    calc_sat_state_almanac(&simulation_almanacs[i], &t,
-      simulation_sats_pos[i], simulation_sats_vel[i],
-      &clock_err, &clock_rate_err);
+    s8 r = calc_sat_state_almanac(&simulation_almanacs[i], &t,
```
Comment (jacobmcnamee) [inline src/simulator.c]: So I guess the simulator will just crash after 6 days...

**Comment by jacobmcnamee** [comment]
lgtm, just update libswiftnav pointer once https://github.com/swift-nav/libswiftnav/pull/345 is merged.

**Pull request merged**